drivers/adc/include/adc/adc.h

```diff
@@ -146,13 +146,15 @@ struct adc_chan_config {
     uint16_t c_refmv;
     uint8_t c_res;
     uint8_t c_configured;
+    uint8_t c_cnum;
 };
 
 struct adc_dev {
     struct os_dev ad_dev;
     struct os_mutex ad_lock;
     struct adc_driver_funcs ad_funcs;
     struct adc_chan_config *ad_chans;
+    void *adc_dev_cfg;
```

Comment:
please keep structure prefixes consistent.  every other element is prefixed with "ad_"